miio/roidmivacuum_miot.py

```diff
@@ -0,0 +1,769 @@
+            return PathMode.Unknown
+
+    @property
+    def mop_present(self) -> bool:
+        """Return True is mop (water box) is installed."""
+        return self.data["mop_present"]
+
+    @property
+    def work_station_frequency(self) -> int:
+        """work_station_freq (2 means base dust colect every second time)."""
+        return self.data["work_station_freq"]
+
+    @property
+    def timing(self) -> str:
+        """Repeated cleaning
+        Example: {"time":[[32400,1,3,0,[1,2,3,4,5],0,[12,10],null],[57600,0,1,2,[1,2,3,4,5,6,0],2,[],null]],"tz":2,"tzs":7200}
+        Cleaning 1:
+            32400 = startTime(9:00)
+            1=Enabled
+            3=FanSpeed.Strong
+            0=SweepType.Sweep
+            [1,2,3,4,5]=Monday-Friday
+            0=WaterLevel
+            [12,10]=List of rooms
+            null: ?Might be related to "Customize"?
+        Cleaning 2:
+            57600 = startTime(16:00)
+            0=Disabled
+            1=FanSpeed.Silent
+            2=SweepType.MopAndSweep
+            [1,2,3,4,5,6,0]=Monday-Sunday
+            2=WaterLevel.Second
+            []=All rooms
+            null: ?Might be related to "Customize"?
+        tz/tzs= time-zone
+        """
+        return self.data["timing"]
+
+    @property
+    def carpet_mode(self) -> bool:
+        """Auto boost on carpet."""
+        return self.data["auto_boost"]
+
+    def _parse_forbid_mode(self, val) -> DNDStatus:
+        # Example data: {"time":[75600,21600,1],"tz":2,"tzs":7200}
+        def _seconds_to_components(val):
+            hour = math.floor(val / 3600)
+            minut = math.floor((val - hour * 3600) / 60)
+            return (hour, minut)
+
+        as_dict = json.loads(val)
+        enabled = bool(as_dict["time"][2])
+        start = _seconds_to_components(as_dict["time"][0])
+        end = _seconds_to_components(as_dict["time"][1])
+        return DNDStatus(
+            dict(
+                enabled=enabled,
+                start_hour=start[0],
+                start_minute=start[1],
+                end_hour=end[0],
+                end_minute=end[1],
+            )
+        )
+
+    @property
+    def dnd_status(self) -> DNDStatus:
+        """Returns do-not-disturb status."""
+        return self._parse_forbid_mode(self.data["forbid_mode"])
+
+    @property
+    def water_level(self) -> WaterLevel:
+        """Get current water level."""
+        try:
+            return WaterLevel(self.data["water_level"])
+        except ValueError:
+            _LOGGER.error("Unknown WaterLevel (%s)", self.data["water_level"])
+            return WaterLevel.Unknown
+
+    @property
+    def double_clean(self) -> bool:
+        """Is double clean enabled."""
+        return self.data["double_clean"]
+
+    @property
+    def edge_sweep(self) -> bool:
+        """Is edge clean enabled."""
+        return self.data["edge_sweep"]
+
+    @property
+    def vacuum_led(self) -> bool:
```

Comment:
```suggestion
    def led(self) -> bool:
```
this will make it consistent with other integrations. It is enough to have the `station_` prefix below for the station led to differentiate it from this.